capa/features/extractors/ida/helpers.py

```diff
@@ -26,15 +27,19 @@ def find_byte_sequence(start: int, end: int, seq: bytes) -> Iterator[int]:
         end: max virtual address
         seq: bytes to search e.g. b"\x01\x03"
     """
+    patterns = ida_bytes.compiled_binpat_vec_t()
+    encoding = ida_nalt.get_default_encoding_idx(ida_nalt.BPU_1B)
+
     seqstr = " ".join([f"{b:02x}" for b in seq])
-    while True:
-        # TODO(mike-hunhoff): find_binary is deprecated. Please use ida_bytes.bin_search() instead.
-        # https://github.com/mandiant/capa/issues/1606
-        ea = idaapi.find_binary(start, end, seqstr, 0, idaapi.SEARCH_DOWN)
-        if ea == idaapi.BADADDR:
-            break
-        start = ea + 1
-        yield ea
+    err = ida_bytes.parse_binpat_str(patterns, 0, seqstr, 16, encoding)
+
+    if not err:
```

Comment:
Let’s change to this to return if `err` is set so we can avoid nesting the following `while` statement.

Reply:
That makes sense!

I have resolved this in c9481a73d710372df6647835db584aa5d81dc861.